include/ginkgo/core/base/exception.hpp

```diff
@@ -54,7 +56,7 @@ namespace gko {
  *
  * @ingroup error
  */
-class Error : public std::exception {
+class Error : public std::runtime_error {
```

Comment:
any reason for that?

Reply:
That was a change I added: std::runtime_error has its own resource management for `what()` that works correctly with C++ exception semantics, std::string doesn't handle all edge cases properly IIRC.